Makefile

```diff
@@ -1,35 +1,61 @@
-.PHONY: get-code-generator generate build clean
+APP = tagger
+PLUGIN = kubectl-tag
 
-PROJECT=github.com/ricardomaraschini/tagger
-GEN_OUTPUT=/tmp/$(PROJECT)/infra/tags
+IMAGE_BUILDER ?= podman
+IMAGE ?= quay.io/rmarasch/tagger
+IMAGE_TAG = $(IMAGE):latest
 
-build:
-	go build -mod vendor -o _output/bin/tagger ./cmd/tagger
-	go build -mod vendor -o _output/bin/kubectl-tag ./cmd/kubectl-tag
+OUTPUT_DIR ?= _output
+OUTPUT_BIN = $(OUTPUT_DIR)/bin
 
+TAGGER_BIN = $(OUTPUT_BIN)/$(APP)
+PLUGIN_BIN = $(OUTPUT_BIN)/$(PLUGIN)
+GEN_BIN = $(OUTPUT_DIR)/code-generator
+
+PROJECT = github.com/ricardomaraschini/$(APP)
+GEN_OUTPUT = /tmp/$(PROJECT)/imagetags
+
+default: build
+
+build: $(APP) $(PLUGIN)
+
+.PHONY: $(APP)
+$(APP):
+	go build -o $(TAGGER_BIN) ./cmd/$(APP)
+
+.PHONY: $(PLUGIN)
+$(PLUGIN):
+	go build -o $(PLUGIN_BIN) ./cmd/$(PLUGIN)
+
+.PHONY: get-code-generator
 get-code-generator:
-	rm -rf _output/code-generator
-	git clone --depth=1                                                     \
-		https://github.com/kubernetes/code-generator.git                \
-		_output/code-generator
+	rm -rf $(GEN_BIN) || true
+	git clone --depth=1 \
+		https://github.com/kubernetes/code-generator.git \
+		$(GEN_BIN)
+
+generate: generate-proto generate-k8s
 
+.PHONY: generate-proto
 generate-proto:
-	protoc --go-grpc_out=paths=source_relative:.				\
-		--go_out=paths=source_relative:.				\
+	protoc --go-grpc_out=paths=source_relative:. \
+		--go_out=paths=source_relative:. \
 		./infra/pb/*.proto
 
+.PHONY: generate-k8s
 generate-k8s:
-	_output/code-generator/generate-groups.sh all                           \
-		$(PROJECT)/infra/tags/v1/gen					\
-		$(PROJECT)                                                      \
-		infra/tags:v1                                                   \
-		--go-header-file _output/code-generator/hack/boilerplate.go.txt \
+	$(GEN_BIN)/generate-groups.sh all \
+		$(PROJECT)/infra/tags/v1/gen \
+		$(PROJECT) \
+		infra/tags:v1
```

Comment:
There is a missing `\` at the end of this line. This is why I like to use these `\` aligned to the right, makes it easier to spot. !!OPINIONATED PEOPLE HAVE STRONG OPINIONATED OPINIONS!!

Reply:
Since I've introduced variables the length of those lines changed, and therefore to align them back again will force changing more lines than intended. For now, let's make it work and make sure the `Makefile` targets are working as intended, and then make it pretty :-)